pkg/config/datafileprojectconfig/entities/entities.go

```diff
@@ -106,9 +106,9 @@ type Rollout struct {
 
 // Integration represents a integration from the Optimizely datafile
 type Integration struct {
-	Key       string `json:"key"`
-	Host      string `json:"host"`
-	PublicKey string `json:"publicKey"`
+	Key       *string `json:"key"`
+	Host      string  `json:"host"`
+	PublicKey string  `json:"publicKey"`
```

Comment:
This is because in golang `string` cannot be null, it is always empty if nothing is assigned to it. In our case we need to make sure that `Key` is provided with every integration in the datafile, if not then client should throw an error. This is a internal struct and client will not be able to access it.